src/interfaces/ecpg/test/preproc/meson.build

```diff
@@ -0,0 +1,85 @@
+
+pgc_files = [
+  'comment',
+  'cursor',
+  'define',
+  'init',
+  'outofscope',
+  'type',
+  'variable',
+  'whenever',
+  'whenever_do_continue',
+]
+
+pgc_executables = []
+
+foreach pgc_file: pgc_files
+  exe_input = custom_target(pgc_file,
+    input: '@0@.pgc'.format(pgc_file),
+    output: '@0@.c'.format(pgc_file),
+    command: [
+      ecpg_exe, 
+      '--regression',
+      '-I' + meson.current_source_dir(),
+      '-I' + meson.current_source_dir() + '/../../include/',
+      '-o', '@OUTPUT@', '@INPUT@'
+    ],
+  )
+  pgc_executables += {'exe_name': pgc_file, 'exe_input': exe_input}
+endforeach
+
+pgc_files_extra = [
+  'array_of_struct',
+  'pointer_to_struct',
+]
+
+foreach pgc_file: pgc_files_extra
+  exe_input = custom_target(pgc_file,
+    input: '@0@.pgc'.format(pgc_file),
+    output: '@0@.c'.format(pgc_file),
+    command: [
+      ecpg_exe, 
+      '--regression',
+      '-I' + meson.current_source_dir(),
+      '-I' + meson.current_source_dir() + '/../../include/',
+      '-c', '-o', '@OUTPUT@', '@INPUT@'
+    ],
+  )
+  pgc_executables += {'exe_name': pgc_file, 'exe_input': exe_input}
+endforeach
+
+exe_input = custom_target('autoprep',
+  input: 'autoprep.pgc',
+  output: 'autoprep.c',
+  command: [
+    ecpg_exe, 
+    '--regression',
+    '-I' + meson.current_source_dir(),
+    '-I' + meson.current_source_dir() + '/../../include/',
+    '-r', 'prepare',
+    '-o', '@OUTPUT@', '@INPUT@'
+  ],
+)
+pgc_executables += {'exe_name': 'autoprep', 'exe_input': exe_input}
+
+exe_input = custom_target('strings',
+  input: 'strings.pgc',
+  output: 'strings.c',
+  command: [
+    ecpg_exe, 
+    '--regression',
+    '-I' + meson.current_source_dir(),
+    '-I' + meson.current_source_dir() + '/../../include/',
+    '-i', '-o', '@OUTPUT@', '@INPUT@'
+  ],
+)
```

Comment:
Why are these not using the loop approach above?

Reply:
That has extra `-i` flag.

Reply:
You can solve that by doing something like
```
pgc_files = [
 ...
  'array_of_struct' : {'extra_flags': ['-c']},
  'pointer_to_struct' : {'extra_flags': ['-c']},
  'autoprep': {'extra_flags': ['-r', 'prepare']},
  'strings': {'extra_flags: ['-i']},
]
```

Reply:
Updated.